server/internal/decoy/decoy.go

```diff
@@ -72,11 +72,16 @@ type decoy struct {
 	rng       *mRand.Rand
 	docCh     chan *pkicache.Entry
 
-	surbETAs   *avl.Tree
+	surbETAs   *queue.PriorityQueue
 	surbStore  map[uint64]*surbCtx
 	surbIDBase uint64
 }
 
+func addPriority(ctx *surbCtx) {
+	now := monotime.Now()
+	ctx.priority = uint64(now + ctx.eta)
```

Comment:
can we ensure that priority is always unique?

Reply:
we don't need to ensure it's unique